pkg/collector/corechecks/cluster/orchestrator/orchestrator.go

```diff
@@ -340,21 +345,45 @@ func (o *OrchestratorCheck) processNodes(sender aggregator.Sender) {
 	}
 	groupID := atomic.AddInt32(&o.groupID, 1)
 
-	messages, err := processNodesList(nodesList, groupID, o.orchestratorConfig, o.clusterID)
+	nodesMessages, clusterModel, err := processNodesList(nodesList, groupID, o.orchestratorConfig, o.clusterID)
 	if err != nil {
 		o.Warnf("Unable to process node list: %s", err) //nolint:errcheck
 		return
 	}
+	sendNodesMetadata(sender, nodesList, nodesMessages, o.clusterID)
 
+	clusterMessage, clusterErr := extractClusterMessage(o.orchestratorConfig, o.clusterID, o.apiClient, groupID, clusterModel)
+	if clusterErr != nil {
+		_ = o.Warnf("Could not collect orchestrator cluster information: %s, will still send nodes information", err)
+		return
+	}
+	sendClusterMetadata(sender, clusterMessage, o.clusterID)
+}
+
+func sendNodesMetadata(sender aggregator.Sender, nodesList []*v1.Node, nodesMessages []model.MessageBody, clusterID string) {
 	stats := orchestrator.CheckStats{
-		CacheHits: len(nodesList) - len(messages),
-		CacheMiss: len(messages),
+		CacheHits: len(nodesList) - len(nodesMessages),
+		CacheMiss: len(nodesMessages),
 		NodeType:  orchestrator.K8sNode,
 	}
 
 	orchestrator.KubernetesResourceCache.Set(orchestrator.BuildStatsKey(orchestrator.K8sNode), stats, orchestrator.NoExpiration)
 
-	sender.OrchestratorMetadata(messages, o.clusterID, forwarder.PayloadTypeNode)
+	sender.OrchestratorMetadata(nodesMessages, clusterID, forwarder.PayloadTypeNode)
+}
+
+func sendClusterMetadata(sender aggregator.Sender, clusterMessage model.MessageBody, clusterID string) {
+	if clusterMessage == nil {
```

Comment:
If we don't want that check here, I can move that one to `extractClusterMessage()` instead. Which makes the control flow easier but also increases the complexity of that function.